Source/WebCore/workers/service/WorkerSWClientConnection.cpp

```diff
@@ -553,6 +553,11 @@ void WorkerSWClientConnection::retrieveRecordResponseBody(BackgroundFetchRecordI
     });
 }
 
+void WorkerSWClientConnection::getServiceWorkerClientPendingMessages(ScriptExecutionContextIdentifier, CompletionHandler<void(Vector<ServiceWorkerClientPendingMessage>&&)>&& completionHandler)
+{
+    completionHandler({ });
```

Comment:
I don't see how SharedWorkers can have this issue, as they cannot come into existence until *after* a document page load occurs, so their link to at least one parent ScriptExecutionContext is already properly established.

Reply:
While a ServiceWorker is handling a fetch event for a main resource, there cannot be a worker or shared worker in existence yet. Those can't happen until the DocumentLoader is at the very least committed.

In other words, inside the fetch event handler for a main resource load, matchAll can't find workers spawned from the new document because they can't exist yet.

(SharedWorkers that exist because of a different Document on the origin can be found, but also can't suffer this problem)

Reply:
It is worth testing to be sure, but the fetch event fired for a worker script fetch should expose the worker ID as `resultingClientId`. The worker is not yet created at this time since the script is not there but you can probably postMessage to it.